### ISSUE TITLE
[question] Is Feature List 12_0 GPU required?

### ISSUE THREAD

**Issue by deton24** [HansKristian-Work/vkd3d-proton#386]
I don't know whether it supports AC Valhalla at all, but Vulkan in this game would be highly appreciated, since it requires Feature List 12_0 which means that the game doesn't work on anything below GTX 9XX and Radeon GCN 2.0.
I wonder if vkd3d-proton requires Feature List 12_0 so it would work as a workaround on such GPUs.

**Comment by HansKristian-Work** [comment]
If the game requires 12_0, vkd3d-proton also requires 12_0. VKD3D_FEATURE_LEVEL=12_0 can usually be forced as an environment variable, since games rarely actually make full use of 12_0 features to work.

**Comment by blckbx** [comment]
Since release of AC:V I'm tinkering with Lutris (latest wine version lutris-5.21-2 and dxvk 1.7.2 (with vk3d-proton 2.0 libs of d3d12.dll) to get it running. The game is dx12 only. There's no chance to run it with DX11.
- Environment variables are set to VKD3D_FEATURE_LEVEL=12_0 and VKD3D_SHADER_DEBUG=none. 
- Command line arguments "-d3d12". 
- Winecfg was modified to d3d12.dll (native, built-in)
Uplay starts up fine, games gets stuck up on splash screen and crash dialogue appears.

> Unhandled exception: page fault on read access to 0x00000018 in 64-bit code (0x000000006f7d5965).
> Register dump:
>  rip:000000006f7d5965 rsp:00000000048eed68 rbp:0000000001807170 eflags:00010202 (  R- --  I   - - - )
>  rax:0000000000000001 rbx:000000002b58fcd0 rcx:0000000000000000 rdx:0000000000000000
>  rsi:000000002b52eda0 rdi:00000000048eee40  r8:0000000000000001  r9:00000000048eeb8c r10:0000000000000000
>  r11:0000000000000002 r12:0000000000000007 r13:0000000000000000 r14:0000000000000000 r15:00000000048eefd0
> Stack dump:
> 0x00000000048eed68:  000000006f8021dd 0000000000000001
> 0x00000000048eed78:  00000000048eedc0 0000000000000000
> 0x00000000048eed88:  0000000000000000 0000000000000000
> 0x00000000048eed98:  0000000000000000 0000000000000000
> 0x00000000048eeda8:  0000000000000000 0000000000000001
> 0x00000000048eedb8:  00000000048eee40 0000000000000000
> 0x00000000048eedc8:  0000000000000000 00000000017469a0
> 0x00000000048eedd8:  0000000000000080 0000000000000200
> 0x00000000048eede8:  000000006f7c29cd 0000000100015800
> 0x00000000048eedf8:  0000000000000010 0000000000302600
> 0x00000000048eee08:  0000008000000080 0000000100000001
> 0x00000000048eee18:  0000000100000000 0000000000000000
> Backtrace:
> =>0 0x000000006f7d5965 EntryPoint+0x14615() in d3d12 (0x0000000001807170)
>   1 0x000000006f8021dd EntryPoint+0x40e8c() in d3d12 (0x0000000001807170)
>   2 0x0000000140895abe EntryPoint+0xe1f18a9d() in acvalhalla (0x00000000048eefe0)
>   3 0x000000014088dc19 EntryPoint+0xe1f10bf8() in acvalhalla (0x0000000000004000)
>   4 0x0000000140c8215f EntryPoint+0xe230513e() in acvalhalla (0x00000000048ef299)
>   5 0x0000000140c4a57a EntryPoint+0xe22cd559() in acvalhalla (0x0000000000000000)
>   6 0x0000000140b86cdc EntryPoint+0xe2209cbb() in acvalhalla (0x0000000000000000)
>   7 0x0000000140ace458 EntryPoint+0xe2151437() in acvalhalla (0x0000000000000000)
>   8 0x0000000140a37ea7 EntryPoint+0xe20bae86() in acvalhalla (0x0000000036aa15c0)
>   9 0x0000000140a1db3f EntryPoint+0xe20a0b1e() in acvalhalla (0x0000000036aa15c0)
>   10 0x000000014070bc1b EntryPoint+0xe1d8ebfa() in acvalhalla (0x00000000048ef660)
>   11 0x000000014076765b EntryPoint+0xe1dea63a() in acvalhalla (0x000000000000001c)
>   12 0x000000014075828a EntryPoint+0xe1ddb269() in acvalhalla (0x00000000034e7c10)
>   13 0x00000001407779a5 EntryPoint+0xe1dfa984() in acvalhalla (0x00000000034e7c10)
>   14 0x000000014078f11c EntryPoint+0xe1e120fb() in acvalhalla (0x0000000000000370)
>   15 0x000000014052bf5a EntryPoint+0xe1baef39() in acvalhalla (0x0000000000000000)
>   16 0x000000014052bcd3 EntryPoint+0xe1baecb2() in acvalhalla (0x0000000000000000)
>   17 0x0000000143e7585c EntryPoint+0xe54f883b() in acvalhalla (0x0000000000000000)
>   18 0x000000007b62ce99 EntryPoint+0xffffffffffffffff() in kernel32 (0x0000000000000000)
>   19 0x000000007bc59f13 EntryPoint+0xffffffffffffffff() in ntdll (0x0000000000000000)
> 0x000000006f7d5965 EntryPoint+0x14615 in d3d12: movl	0x0000000000000018(%rcx),%edx
> Modules:
> Module	Address					Debug info	Name (92 modules)
> PE	          230000-          68c000	Deferred        d3dcompiler_47
> PE	          690000-          6e1000	Deferred        shlwapi
> PE	          6f0000-          82d000	Deferred        gfsdk_aftermath_lib.x64
> PE	          830000-          8c1000	Deferred        msvcp140
> PE	          8d0000-          8e9000	Deferred        vcruntime140
> PE	          8f0000-          8f8000	Deferred        api-ms-win-crt-heap-l1-1-0
> PE	          900000-          908000	Deferred        api-ms-win-crt-utility-l1-1-0
> PE	          910000-          97a000	Deferred        dbghelp
> PE	          980000-          fd7000	Deferred        tobii_gameintegration_x64
> PE	          fe0000-         1048000	Deferred        uplay_r2_loader64
> PE	         1050000-         1108000	Deferred        winmm
> PE	         1110000-         1137000	Deferred        wintrust
> PE	         1840000-         1d52000	Deferred        overlay64
> PE	         1e70000-         1ef8000	Deferred        dbdata
> PE	         1f00000-         2295000	Deferred        uplay_aux_r164
> PE	         23b0000-         2852000	Deferred        uplay_r264
> PE	        61540000-        6157f000	Deferred        advapi32
> PE	        61d00000-        61d09000	Deferred        api-ms-win-crt-convert-l1-1-0
> PE	        61d80000-        61d8a000	Deferred        api-ms-win-crt-math-l1-1-0
> PE	        62240000-        624d1000	Deferred        d3d9
> PE	        62980000-        62988000	Deferred        api-ms-win-core-localization-l1-2-1
> PE	        62b40000-        62b47000	Deferred        api-ms-win-crt-environment-l1-1-0
> PE	        62c80000-        62cba000	Deferred        wbemprox
> PE	        62dc0000-        62e42000	Deferred        rpcrt4
> PE	        63280000-        6328d000	Deferred        version
> PE	        63400000-        63449000	Deferred        sapi
> PE	        63540000-        63549000	Deferred        api-ms-win-crt-runtime-l1-1-0
> PE	        639c0000-        639d9000	Deferred        shcore
> PE	        64840000-        648c5000	Deferred        gdiplus
> PE	        64cc0000-        64df5000	Deferred        oleaut32
> PE	        65380000-        6538e000	Deferred        wtsapi32
> PE	        65b40000-        65b48000	Deferred        api-ms-win-crt-filesystem-l1-1-0
> PE	        66040000-        66056000	Deferred        bcrypt
> PE	        66400000-        66408000	Deferred        api-ms-win-crt-locale-l1-1-0
> PE	        664c0000-        664cf000	Deferred        hid
> PE	        66540000-        66552000	Deferred        rtworkq
> PE	        66c40000-        66c48000	Deferred        api-ms-win-appmodel-runtime-l1-1-2
> PE	        67600000-        67607000	Deferred        api-ms-win-core-string-l1-1-0
> PE	        67b40000-        67b66000	Deferred        mmdevapi
> PE	        67e80000-        67ea0000	Deferred        jsproxy
> PE	        68040000-        68092000	Deferred        mfplat
> PE	        68300000-        68352000	Deferred        combase
> PE	        68500000-        6852d000	Deferred        uxtheme
> PE	        68c00000-        68c13000	Deferred        aclui
> PE	        69200000-        6920c000	Deferred        vcruntime140_1
> PE	        69400000-        69443000	Deferred        winhttp
> PE	        69740000-        6976a000	Deferred        propsys
> PE	        698c0000-        698cf000	Deferred        vulkan-1
> PE	        6a200000-        6a317000	Deferred        ole32
> PE	        6a840000-        6a87c000	Deferred        usp10
> PE	        6aa80000-        6aa87000	Deferred        api-ms-win-core-fibers-l1-1-1
> PE	        6ac00000-        6ac2d000	Deferred        mfreadwrite
> PE	        6b300000-        6b307000	Deferred        api-ms-win-core-localization-obsolete-l1-2-0
> PE	        6b5c0000-        6b5c9000	Deferred        api-ms-win-crt-stdio-l1-1-0
> PE	        6b680000-        6b68f000	Deferred        xinput9_1_0
> PE	        6ba00000-        6ba29000	Deferred        sechost
> PE	        6bac0000-        6bb34000	Deferred        setupapi
> PE	        6c0c0000-        6c0d2000	Deferred        pdh
> PE	        6c7c0000-        6c90e000	Deferred        gdi32
> PE	        6ca40000-        6cb84000	Deferred        comctl32
> PE	        6cd40000-        6cd47000	Deferred        api-ms-win-crt-time-l1-1-0
> PE	        6d940000-        6d94f000	Deferred        imagehlp
> PE	        6dc80000-        6dc88000	Deferred        api-ms-win-core-synch-l1-2-0
> PE	        6eb00000-        6ed24000	Deferred        user32
> PE	        6ef40000-        6ef48000	Deferred        api-ms-win-appmodel-runtime-l1-1-1
> PE	        6f200000-        6f3e1000	Deferred        dxgi
> PE	        6f7c0000-        6f9f5000	Export          d3d12
> PE	        6fc40000-        6fc72000	Deferred        rsaenh
> PE	        6fd00000-        6fd09000	Deferred        api-ms-win-crt-string-l1-1-0
> PE	        6fd80000-        6fd87000	Deferred        api-ms-win-core-datetime-l1-1-1
> PE	        700c0000-        700d5000	Deferred        api-ms-win-crt-private-l1-1-0
> PE	        70b00000-        70bff000	Deferred        comdlg32
> PE	        71000000-        71018000	Deferred        imm32
> PE	        7b000000-        7b0d5000	Deferred        kernelbase
> PE	        7b600000-        7b811000	Export          kernel32
> PE	        7bc00000-        7bc9a000	Export          ntdll
> PE	       140000000-       15f6e3000	Export          acvalhalla
> PE	       180000000-       180025000	Deferred        amd_ags_x64
> PE	    7f52e00b0000-    7f52e00bb000	Deferred        dinput8
> PE	    7f52e01b0000-    7f52e01b4000	Deferred        xaudio2_9
> PE	    7f52e04b0000-    7f52e04b3000	Deferred        winepulse
> PE	    7f52e0950000-    7f52e0956000	Deferred        windowscodecs
> PE	    7f52e0c90000-    7f52e0c9c000	Deferred        winspool
> PE	    7f52e1530000-    7f52e1534000	Deferred        winex11
> PE	    7f52e1950000-    7f52e1954000	Deferred        ws2_32
> PE	    7f52e19a0000-    7f52e2296000	Deferred        shell32
> PE	    7f52e23f0000-    7f52e23f4000	Deferred        iphlpapi
> PE	    7f52e2440000-    7f52e2444000	Deferred        msvcrt
> PE	    7f52e2500000-    7f52e2504000	Deferred        winevulkan
> PE	    7f52e2580000-    7f52e25c4000	Deferred        crypt32
> PE	    7f52e26e0000-    7f52e26e4000	Deferred        ucrtbase
> PE	    7f52e28e0000-    7f52e28e3000	Deferred        wow64cpu
> Threads:
> process  tid      prio (all id:s are in hex)
> 00000038 services.exe
> 	["C:\windows\system32\services.exe"]
> 	0000003c    0
> 	00000040    0
> 	0000004c    0
> 	0000005c    0
> 	0000006c    0
> 	00000084    0
> 	00000090    0
> 	000000b4    0
> 	000000dc    0
> 00000044 winedevice.exe
> 	[C:\windows\system32\winedevice.exe]
> 	00000048    0
> 	00000054    0
> 	00000058    0
> 	00000060    0
> 00000064 plugplay.exe
> 	[C:\windows\system32\plugplay.exe]
> 	00000068    0
> 	00000070    0
> 	00000074    0
> 	00000078    0
> 	00000098    0
> 	0000009c    0
> 	000004f4    0
> 	00000664    0
> 	0000027c    0
> 	00000360    0
> 	00000400    0
> 	000001e4    0
> 	000002a0    0
> 	0000045c    0
> 	000002d4    0
> 	00000494    0
> 	00000404    0
> 	00000320    0
> 	000002d8    0
> 	000005c8    0
> 	00000678    0
> 	0000044c    0
> 	00000324    0
> 	000004fc    0
> 	000003dc    0
> 0000007c winedevice.exe
> 	[C:\windows\system32\winedevice.exe]
> 	00000080    0
> 	00000088    0
> 	0000008c    0
> 	00000094    0
> 	000000a0    0
> 	000000a8    0
> 000000ac svchost.exe
> 	[C:\windows\system32\svchost.exe -k LocalServiceNetworkRestricted]
> 	000000b0    0
> 	000000b8    0
> 	000000bc    0
> 000000d4 rpcss.exe
> 	[C:\windows\system32\rpcss.exe]
> 	000000d8    0
> 	000000e0    0
> 	000000e4    0
> 	000000e8    0
> 	000000ec    0
> 	00000568    0
> 	000001f0    0
> 00000558 explorer.exe
> 	[C:\windows\system32\explorer.exe /desktop]
> 	0000055c    0
> 	00000560    0
> 	00000564    0
> 000001a8 explorer.exe
> 	[C:\windows\system32\explorer.exe /desktop]
> 	00000594    0
> 	00000024    0
> 	0000057c    0
> 00000598 upc.exe
> 	["C:\Program Files (x86)\Ubisoft\Ubisoft Game Launcher\upc.exe" -d3d12 ]
> 	000007d0    0
> 	00000130    0
> 	00000500    0
> 	000003bc    0
> 	00000454    0
> 	00000118    0
> 	000007dc    0
> 	00000100    0
> 	000007e8    0
> 	000007e4    0
> 	000007f0    0
> 	00000034    0
> 	00000668    0
> 	000005fc    0
> 	00000304    0
> 	00000240   -2
> 	000005e4   -2
> 	000004f0    0
> 	00000350    0
> 	00000520    0
> 	00000528    0
> 	000002fc    0
> 	00000210    0
> 	00000214   -2
> 	000001ac    0
> 	00000198    0
> 	00000190    0
> 	000002f4    0
> 	0000012c    0
> 	00000550   -2
> 	00000524    0
> 	00000488    0
> 	000001e0    0
> 	0000054c    0
> 	000003fc    0
> 	00000470    0
> 	000004bc    0
> 	00000504    0
> 	000002c4    0
> 	0000028c   -2
> 	00000798    0
> 	000006d4    0
> 	000000c0    0
> 	00000618    0
> 000004a0 UplayWebCore.exe
> 	["C:/Program Files (x86)/Ubisoft/Ubisoft Game Launcher/UplayWebCore.exe" --type=renderer --no-sandbox --disable-features=TouchpadAndWheelScrollLatching --service-pipe-token=1B0CA98258B8DE088BDCE1DB8143C703 --lang=en-US --locales-dir-path="C:/Program Files (x86)/Ubisoft/Ubisoft Game Launcher/locales/1/" --log-file="C:\Program Files (x86)\Ubisoft\Ubisoft Game Launcher\debug.log" --disable-spell-checking --enable-system-flash --device-scale-factor=1 --num-raster-threads=4 --enable-main-frame-before-activation --service-request-channel-token=1B0CA98258B8DE088BDCE1DB8143C703 --renderer-client-id=4 --mojo-platform-channel-handle=1192 /prefetch:1]
> 	0000065c    0
> 	00000720    0
> 	000004ec    0
> 	00000278   -2
> 	00000164   -2
> 	000003a8    0
> 	00000174    0
> 	000004b0    0
> 	00000700    0
> 	0000050c    0
> 	000004e8    0
> 	00000128    0
> 	00000590    0
> 	00000414    0
> 	000006e8    0
> 	0000059c    0
> 	000000f0   -2
> 	0000060c    0
> 	00000554    0
> 	00000030    0
> 	00000574    0
> 	00000338    0
> 	000006ac    0
> 	00000628    0
> 	00000620    0
> 	00000460    0
> 	00000748    0
> 	00000690    0
> 	00000718    0
> 	00000738    0
> 	00000108    0
> 000007cc UplayWebCore.exe
> 	["C:/Program Files (x86)/Ubisoft/Ubisoft Game Launcher/UplayWebCore.exe" --type=renderer --no-sandbox --disable-features=TouchpadAndWheelScrollLatching --disable-gpu-compositing --service-pipe-token=094F0BB95B17213D766BC6975A113439 --lang=en-US --locales-dir-path="C:/Program Files (x86)/Ubisoft/Ubisoft Game Launcher/locales/1/" --log-file="C:\Program Files (x86)\Ubisoft\Ubisoft Game Launcher\debug.log" --disable-spell-checking --enable-system-flash --device-scale-factor=1 --num-raster-threads=4 --enable-main-frame-before-activation --service-request-channel-token=094F0BB95B17213D766BC6975A113439 --renderer-client-id=6 --mojo-platform-channel-handle=1512 /prefetch:1]
> 	000007c0    0
> 	00000814    0
> 	00000810    0
> 	000007fc   -2
> 	0000080c   -2
> 	00000708    0
> 	000004e4    0
> 	000005bc    0
> 	00000584    0
> 	0000040c    0
> 	00000390    0
> 	000003a0    0
> 	000005b4    0
> 	00000234    0
> 	00000790    0
> 	000004d4    0
> 	000006bc   -2
> 	00000238    0
> 	000001cc    0
> 	00000800    0
> 	000004c8    0
> 	000004a4    0
> 	000006f0    0
> 	0000010c    0
> 	00000398    0
> 	00000614    0
> 	00000670    0
> 0000037c UplayWebCore.exe
> 	["C:/Program Files (x86)/Ubisoft/Ubisoft Game Launcher/UplayWebCore.exe" --type=renderer --no-sandbox --disable-features=TouchpadAndWheelScrollLatching --disable-gpu-compositing --service-pipe-token=D805E38A16BD9E60396AC1AA9C1A4896 --lang=en-US --locales-dir-path="C:/Program Files (x86)/Ubisoft/Ubisoft Game Launcher/locales/1/" --log-file="C:\Program Files (x86)\Ubisoft\Ubisoft Game Launcher\debug.log" --disable-spell-checking --enable-system-flash --device-scale-factor=1 --num-raster-threads=4 --enable-main-frame-before-activation --service-request-channel-token=D805E38A16BD9E60396AC1AA9C1A4896 --renderer-client-id=7 --mojo-platform-channel-handle=1536 /prefetch:1]
> 	00000818    0
> 	0000075c    0
> 	00000120    0
> 	000001bc   -2
> 	00000750   -2
> 	000003a4    0
> 	00000638    0
> 	00000418    0
> 	00000268    0
> 	000003b8    0
> 	0000002c    0
> 	000002e8    0
> 	0000014c    0
> 	000007b4    0
> 	00000780    0
> 	000000f4    0
> 	00000440   -2
> 	00000050    0
> 	00000608    0
> 	00000570    0
> 	000004d8    0
> 	000005b8    0
> 	00000604    0
> 	0000061c    0
> 000002ec UbisoftGameLauncher.exe
> 	["C:\Program Files (x86)\Ubisoft\Ubisoft Game Launcher\UbisoftGameLauncher.exe" -gamelauncher_wait_handle 188 -upc_uplay_id 13504 -upc_game_version 1 -upc_exe_path QzpcUHJvZ3JhbSBGaWxlcyAoeDg2KVxVYmlzb2Z0XEFzc2Fzc2luJ3MgQ3JlZWQgVmFsaGFsbGFcQUNWYWxoYWxsYS5leGU= -upc_working_directory QzpcUHJvZ3JhbSBGaWxlcyAoeDg2KVxVYmlzb2Z0XEFzc2Fzc2luJ3MgQ3JlZWQgVmFsaGFsbGE= -upc_arguments  ]
> 	000002f0    0
> 	00000588    0
> 	00000688    0
> 	000005e8    0
> 000003b4 (D) C:\Program Files (x86)\Ubisoft\Assassin's Creed Valhalla\ACValhalla.exe
> 	["C:/Program Files (x86)/Ubisoft/Assassin's Creed Valhalla/ACValhalla.exe"]
> 	0000058c    0
> 	0000026c    0
> 	00000480    1
> 	0000076c    1
> 	0000018c    1
> 	000003d4   -2 <==
> 	000002c0   -2
> 	00000358   -2
> 	0000071c   -2
> 	00000138    0
> 	000003c8    0
> 	000005ec    0
> 	00000730    0
> 	00000188    0
> 	00000450    0
> 	00000484    0
> 	0000056c    0
> 	0000062c    0
> 	00000674    1
> 	00000410    0
> 	00000228    1
> 	00000508    0
> 	000004f8    0
> 	00000394    1
> 	00000300    0
> 	00000448    0
> 	0000070c   15
> 	000004c0    0
> 	000000fc    0
> 	000007f8    0
> 	000003c0  -15
> 	000002bc    0
> 	000002dc    0
> 	000003f4    0
> 	0000041c  -15
> 	0000036c    0
> System information:
>     Wine build: wine-5.21 (Staging)
>     Platform: x86_64
>     Version: Windows 10
>     Host system: Linux
>     Host version: 5.8.0-28-generic

**Comment by doitsujin** [comment]
vkd3d-proton can be used in any wine environment.

Anyway, the game is known not to work, it's just that we haven't really had time to look into it yet and the game not being on Steam doesn't exactly make things easier. We'll get to it eventually.

**Comment by blckbx** [comment]
@doitsujin If I can be of any help or contribution, just give me a shout. Thanks for all your hard work here!

**Comment by shelterx** [comment]
I'm sure you know AC Valhalla doesn't run well on Nvidia cards, I thought I had it working but now I just freeze in the exact same spot. Not sure if it's an vkd3d problem, wine problem or driver problem.

This always seem to happen:
`0680:err:sync:RtlpWaitForCriticalSection section 0000000005BF0B50 "?" wait timed out in thread 0680, blocked by 0694, retrying (60 sec)`

**Comment by K0bin** [comment]
@shelterx Open a new bug report. This is unrelated to the question.

**Comment by Alecton4** [comment]
Off topic, but may I ask where is the documentation of `VKD3D_FEATURE_LEVEL` variable? I did not find it in README.

**Comment by Blisto91** [comment]
It isn't documented per say besides [it's code](https://github.com/HansKristian-Work/vkd3d-proton/blob/0c962c4758eace2491a6bb0cf3bcdde9cd240f34/libs/vkd3d/device.c#L7427) or original PR https://github.com/HansKristian-Work/vkd3d-proton/pull/93
